es-core/src/GuiComponent.cpp

```diff
@@ -320,6 +347,11 @@ void GuiComponent::applyTheme(const std::shared_ptr<ThemeData>& theme, const std
 
 	if(properties & ThemeFlags::SIZE && elem->has("size"))
 		setSize(elem->get<Eigen::Vector2f>("size").cwiseProduct(scale));
+
+	if(properties & ThemeFlags::Z_INDEX && elem->has("zIndex"))
+		setZIndex(elem->get<float>("zIndex"));
+	else
```

Comment:
I did not actually test this, but in its current form, I don't think this will work, as the getDefaultZIndex() function will just return the current z-index, not retrieve a true default value when needed.  How can it work?